### PR TITLE
Add a new async search security origin

### PR DESCRIPTION
**Pull request by albertzaharovits**
Creates a new security origin, and an associated reserved user and role `_async_search`, which can be used by internal clients to manage the `.async-search-*` restricted index namespace.

_I think we will also need to create a new origin for ILM since the `.async-search-*` restricted indices are managed by ILM and ILM uses the `xpack` security origin which doesn't have restricted indices privileges._

CC @jimczi
Relates #49931 #50294

### CODE REVIEW

**Comment by albertzaharovits** [comment]
@elasticmachine run elasticsearch-ci/bwc

File: x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authz/AuthorizationUtilsTests.java [hunk 1]
```diff
@@ -95,9 +97,15 @@ public void testSwitchAndExecuteXpackSecurityUser() throws Exception {
     }
 
     public void testSwitchAndExecuteXpackUser() throws Exception {
-        String origin = randomFrom(ClientHelper.ML_ORIGIN, ClientHelper.WATCHER_ORIGIN, ClientHelper.DEPRECATION_ORIGIN,
-                ClientHelper.MONITORING_ORIGIN, ClientHelper.PERSISTENT_TASK_ORIGIN, ClientHelper.INDEX_LIFECYCLE_ORIGIN);
-        assertSwitchBasedOnOriginAndExecute(origin, XPackUser.INSTANCE);
+        for (String origin : Arrays.asList(ClientHelper.ML_ORIGIN, ClientHelper.WATCHER_ORIGIN, ClientHelper.DEPRECATION_ORIGIN,
```
Comment (tvernum) [inline x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authz/AuthorizationUtilsTests.java]: 👍

File: x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authz/store/CompositeRolesStoreTests.java [hunk 2]
```diff
@@ -959,13 +960,23 @@ public void testDoesNotUseRolesStoreForXPackUser() {
...
         PlainActionFuture<Role> rolesFuture = new PlainActionFuture<>();
         Authentication auth = new Authentication(XPackUser.INSTANCE, new RealmRef("name", "type", "node"), null);
         compositeRolesStore.getRoles(XPackUser.INSTANCE, auth, rolesFuture);
-        final Role roles = rolesFuture.actionGet();
+        Role roles = rolesFuture.actionGet();
         assertThat(roles, equalTo(XPackUser.ROLE));
         assertThat(effectiveRoleDescriptors.get(), is(nullValue()));
         verifyNoMoreInteractions(fileRolesStore, nativeRolesStore, reservedRolesStore);
+
+        // test AyncSearch user short circuits to its onw reserved role
```
Comment (tvernum) [inline x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authz/store/CompositeRolesStoreTests.java]:
```suggestion
        // test AyncSearch user short circuits to its own reserved role
```

**Review by tvernum — APPROVED** [review]
LGTM

**Comment by albertzaharovits** [comment]
> I think we will also need to create a new origin for ILM since the .async-search-* restricted indices are managed by ILM and ILM uses the xpack security origin which doesn't have restricted indices privileges.

This is not required. Looks like the async search pllugin does not use ILM managed indices anymore. Even if it would, the ILM security origin must never be granted privileges over restricted indices because that would allow any user with privileges for ILM to interfere with restricted indices, even if it does not have privileges over restricted indices.

**Pull request merged**